### DISCUSSION TITLE
module 'whisper' has no attribute 'load_model'

### DISCUSSION THREAD

**Discussion by feitgemel** [openai/whisper#143] [Q&A] [answered]
Hi,
Try this simple python code after the Whisper setup:

import whisper
model = whisper.load_model("base")

and get this error : "AttributeError: module 'whisper' has no attribute 'load_model'"

The whisper command CLI , works fine . This error is if I try to run in IDE

Can you help ?

Eran

**Comment by jongwook** [comment, marked as answer]
It seems possible that your IDE is configured for a different Python installation/environment from what was used in the command line, and it somehow installed the unrelated [whisper package](https://pypi.org/project/whisper/) from PyPI. You can confirm this by:

```python
import whisper
print(whisper.__file__)
```

If the printed path ends with `whisper.py`, you're using the wrong package; please reinstall the package by running:

```bash
pip uninstall whisper
pip install git+https://github.com/openai/whisper.git 
```

**Reply by Jackson214** [reply]
It's giving me the same error in a venv. The error is thrown when importing whisper. Trying to figure this out on my end too, good luck everyone!

**Reply by feitgemel** [reply]
Hello Jongwook,
Thank you very much.

Uninstall and re-install solved the problem.

Eran

**Reply by oncebeloved** [reply]
所以如何解决这个问题呢，我在(whisper) once@ubuntu:~/whisper 虚拟环境中安装了它，但是import whisper失败
![image](https://user-images.githubusercontent.com/51181996/225896071-8f66f83e-c915-475a-a8d7-862181678e5d.png)

**Reply by Jackson214** [reply]
For those still having this issue:
Your problem is probably that you named your file "whisper.py", try naming it something else.

**Reply by bsgxiaobi** [reply]
Thank you.you are right,I have this problem

**Reply by a2nath** [reply]
This was the problem for me

**Reply by pedro380085** [reply]
Thank you, such a dumb mistake! called whisperA.py now.

**Reply by shankarkarande** [reply]
Thank you so much, It works very well for me also !

**Comment by Norlandz** [comment]
Another possible problem: 
Dont name your script file as `whisper.py`. (silly mistake by me...)

**Comment by danielmlow** [comment]
The above didn't work, but this did:

```
pip uninstall whisper
pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git
```

You delete any `whisper` dir or `whisper.py` file if they exist in `site-packages` where you're trying to install (e.g., `miniconda3/envs/environment_name/lib/python3.12/site-packages/`)

**Comment by tejastank** [comment]
I face same issue 

pip install git+https://github.com/openai/whisper.git

Worked perfectly.

Then load_model started to works.

import whisper
model = whisper.load_model("base")

Thanks,
tejas@xamta.in

**Comment by ccortezmunoz** [comment]
Hi, it was not working for me because it was crashing the installation of whisper in python 3.12 and 3.10, I deleted python 3.12, installed whisper and dependencies again and managed to run the script without errors.

**Comment by trevorprater** [comment]
You probably named your working file *whisper.py*.
